pkg/agent/util/net_linux.go

```diff
@@ -376,3 +378,148 @@ func renameHostInterface(oriName string, newName string) error {
+			if err := EnsureIPv6EnabledOnInterface(interfaceName); err != nil {
+				klog.ErrorS(err, "Failed to ensure that IPv6 is enabled on bridge local port, moving uplink IPs to bridge is likely to fail", "port", interfaceName)
+			}
+			break
+		}
+	}
+
+	if err = ConfigureLinkAddresses(localLink.Attrs().Index, interfaceIPs); err != nil {
+		return "", false, err
+	}
+	if err = ConfigureLinkAddresses(interfaceIndex, nil); err != nil {
+		return "", false, err
+	}
+	// Restore the host routes which are lost when moving the network configuration of the
+	// host interface to OVS bridge interface.
+	if err = ConfigureLinkRoutes(localLink, interfaceRoutes); err != nil {
+		return "", false, err
+	}
+	return bridgedName, false, nil
+}
+
+// RestoreHostInterfaceConfiguration restore the configuration from bridge back to host interface, reverting the
+// actions taken in PrepareHostInterfaceConnection.
+func RestoreHostInterfaceConfiguration(brName string, interfaceName string) {
+	klog.InfoS("Restoring bridge config to host interface")
+	if interfaceName != "" {
+		bridgedName := GenerateUplinkInterfaceName(interfaceName)
+		// restore if interface eth0~ exists
+		if exists, err := interfaceExists(bridgedName); exists || err != nil {
+			// get interface config
+			var interfaceIPs []*net.IPNet
+			var interfaceRoutes []interface{}
+			if exists, err = interfaceExists(interfaceName); exists || err != nil {
+				_, interfaceIPs, interfaceRoutes, err = GetInterfaceConfig(interfaceName)
+				if err != nil {
+					klog.ErrorS(err, "Failed to get interface config", "interface", interfaceName)
+				}
+			}
+
+			// remove host interface (eth0~) from bridge
+			if err = deleteOVSPort(brName, bridgedName); err != nil {
+				klog.ErrorS(err, "Delete OVS port failed", "port", bridgedName)
+			}
+
+			// delete internal port (eth0)
+			if err = deleteOVSPort(brName, interfaceName); err != nil {
+				klog.ErrorS(err, "Delete OVS port failed", "port", bridgedName)
+			}
+
+			// rename host interface(eth0~ -> eth0)
+			if err = RenameInterface(bridgedName, interfaceName); err != nil {
+				klog.ErrorS(err, "Restore host interface name failed", "from", bridgedName, "to", interfaceName)
+			} else {
+				var link netlink.Link
+				if link, err = netlink.LinkByName(interfaceName); err != nil {
+					klog.ErrorS(err, "Failed to get link", "interface", interfaceName)
+				} else {
+					if len(interfaceIPs) > 0 {
+						// restore IPs to eth0
+						if err = ConfigureLinkAddresses(link.Attrs().Index, interfaceIPs); err != nil {
+							klog.ErrorS(err, "Restore IPs to host interface failed", "interface", interfaceName)
+						}
+					}
+					if len(interfaceRoutes) > 0 {
+						// restore routes to eth0
+						if err = ConfigureLinkRoutes(link, interfaceRoutes); err != nil {
+							klog.ErrorS(err, "Restore routes to host interface failed", "interface", interfaceName)
+						}
+					}
+				}
+			}
+			klog.InfoS("Finished to restore bridge config to host interface")
```

Comment:
Could we log bridge name and interface name too?

Reply:
We should return at some errors and do not log this message which sounds like restoration succeeded.